app/views/admin/loans/content/_form.html.slim

```diff
@@ -48,44 +59,68 @@
-    = f.input_field :primary_agent_id, collection: @people, include_blank: true
+    .view-element
+      - if @loan.primary_agent && person_policy(@loan.primary_agent).show?
+        = link_to @loan.primary_agent.name, admin_person_path(@loan.primary_agent)
+      - else
+        = @loan.primary_agent.try(:name)
+    .form-element
+      - if !@loan.primary_agent || person_policy(@loan.primary_agent).show?
+        = f.input_field :primary_agent_id, collection: @agent_choices, include_blank: true
+      - else
+        = @loan.primary_agent.try(:name)
 
   = f.input :secondary_agent
-    .view-element = @loan.secondary_agent.try(:name)
-    = f.input_field :secondary_agent_id, collection: @people, include_blank: true
+    .view-element
+      - if @loan.secondary_agent && person_policy(@loan.secondary_agent).show?
+        = link_to @loan.secondary_agent.name, admin_person_path(@loan.secondary_agent)
+      - else
+        = @loan.secondary_agent.try(:name)
+    .form-element
+      - if !@loan.secondary_agent || person_policy(@loan.secondary_agent).show?
+        = f.input_field :secondary_agent_id, collection: @agent_choices, include_blank: true
+      - else
+        = @loan.secondary_agent.try(:name)
 
   = f.input :length_months
-    .view-element = loan.length_months
+    .view-element = @loan.length_months
     = f.input_field :length_months
 
   = f.input :rate
-    .view-element = loan.rate
+    .view-element = @loan.rate
     = f.input_field :rate
 
   = f.input :signing_date
-    .view-element = loan.signing_date
+    .view-element = @loan.signing_date
     = f.input_field :signing_date
 
   = f.input :first_payment_date
-    .view-element = loan.first_payment_date
+    .view-element = @loan.first_payment_date
     = f.input_field :first_payment_date
 
   = f.input :first_interest_payment_date
-    .view-element = loan.first_interest_payment_date
+    .view-element = @loan.first_interest_payment_date
     = f.input_field :first_interest_payment_date
 
   = f.input :target_end_date
-    .view-element = loan.target_end_date
+    .view-element = @loan.target_end_date
     = f.input_field :target_end_date
 
   = f.input :projected_return
-    .view-element = loan.projected_return
+    .view-element = @loan.projected_return
     = f.input_field :projected_return
 
   = f.input :representative
-    .view-element = @loan.representative.try(:name)
-    = f.input_field :representative_id, collection: @representative_choices, include_blank: true
+    .view-element
+      - if @loan.representative && person_policy(@loan.representative).show?
+        = link_to @loan.representative.name, admin_person_path(@loan.representative)
+      - else
+        = @loan.representative.try(:name)
+    .form-element
+      - if !@loan.representative || person_policy(@loan.representative).show?
+        = f.input_field :representative_id, collection: @representative_choices, include_blank: true
+      - else
+        = @loan.representative.try(:name)
```

Comment:
You asked about abstracting this out... I would use a helper method called e.g. `authorized_form_field`. You only need 3 parameters: the object, the field name (e.g. representative_id), and the choice collection. The policy class can be guessed from the object class, and the link path can be obtained with `url_for(object)`. Make sense?